src/main/java/org/jasig/portlet/announcements/mvc/servlet/RssFeedController.java

```diff
@@ -20,122 +20,211 @@
-    this.announcementService = announcementService;
-  }
+    private static final String PATH_ATTRIBUTE = "path";
 
-  @Override
-  protected ModelAndView handleRequestInternal(
-      HttpServletRequest request, HttpServletResponse response)
-      throws IllegalStateException, IOException {
+    private IAnnouncementService announcementService;
 
-    response.setContentType(CONTENT_TYPE);
+    @Value("${RssFeedController.portalContextName:uPortal}")
+    private String portalContextName;
 
-    Long topicId;
-    try {
-      topicId = Long.valueOf(ServletRequestUtils.getIntParameter(request, "topic"));
-      if (topicId == null) {
-        throw new IllegalStateException("Must specify the topic id");
-      }
-    } catch (Exception e) {
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Must specify topic id");
-      return null;
-    }
+    @Value("${RssFeedController.announcementsPortletFname:announcements}")
+    private String announcementsPortletFname;
 
-    Topic t = null;
-    try {
-      t = announcementService.getTopic(topicId);
-    } catch (PortletException e) {
-      e.printStackTrace();
-    }
+    private ObjectMapper objectMapper = new ObjectMapper();
 
-    if (t == null) {
-      response.sendError(HttpServletResponse.SC_NOT_FOUND, "no such topic");
-      return null;
-    }
+    private final Logger logger = LoggerFactory.getLogger(getClass());
 
-    if (!t.isAllowRss()) {
-      response.sendError(HttpServletResponse.SC_FORBIDDEN, "This topic is not available as RSS");
-      return null;
+    @Autowired
+    public void setAnnouncementService(IAnnouncementService announcementService) {
+        this.announcementService = announcementService;
     }
 
-    // fetch and sort the announcements
-    List<Announcement> announcements = new ArrayList<Announcement>();
-    announcements.addAll(t.getPublishedAnnouncements());
-
-    Collections.sort(announcements);
-
-    // create the feed
-    SyndFeed feed = new SyndFeedImpl();
-    feed.setFeedType("rss_2.0");
-    feed.setTitle(t.getTitle());
-    feed.setLink(request.getRequestURL().append("?topic=").append(topicId.toString()).toString());
-    feed.setDescription(t.getDescription());
-
-    List<SyndEntry> entries = new ArrayList<SyndEntry>();
-    SyndEntry entry;
-    SyndContent description;
-
-    for (Announcement a : announcements) {
-      entry = new SyndEntryImpl();
-      entry.setTitle(a.getTitle());
-      entry.setAuthor(a.getAuthor());
-      if (a.getLink() != null) entry.setLink(a.getLink());
-      entry.setPublishedDate(a.getStartDisplay());
-      description = new SyndContentImpl();
-      description.setType("text/plain");
-      description.setValue(a.getMessage());
-      entry.setDescription(description);
-      entries.add(entry);
+    @RequestMapping(produces = CONTENT_TYPE)
+    public void getRssFeed(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
This method appears to be broken into three logical parts: a part to get topic, a part to generate entries, and a part to assemble that information into a response.
:question: Would there be value in breaking this 128 line function block into three functions?

Reply:
I was thinking the same thing as I updated it.

The issue is all the different HTTP responses the method might send based on the different failure scenarios.  I _don't_ feel that those responses have any business being sprinkled into a bunch of little methods (would make it harder to maintain).

This item can be improved through refactoring, but then there's the issue of timeboxing.